pkg/trtl/replica/sync_test.go

```diff
@@ -0,0 +1,133 @@
+package replica_test
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/rotationalio/honu"
+	"github.com/rotationalio/honu/options"
+	"github.com/stretchr/testify/require"
+	"github.com/trisacrypto/directory/pkg/trtl/config"
+	"github.com/trisacrypto/directory/pkg/trtl/peers/v1"
+	"github.com/trisacrypto/directory/pkg/trtl/replica"
+	"github.com/trisacrypto/directory/pkg/utils/wire"
+	"google.golang.org/protobuf/proto"
+)
+
+func TestSelectZeroOrOnePeer(t *testing.T) {
+	fixtures := loadFixtures(t)
+	testCases := []struct {
+		name     string
+		peers    []string
+		self     string
+		expected string
+	}{
+		{name: "None", peers: []string{}, self: "raphael", expected: ""},
+		{name: "OnlySelf", peers: []string{"raphael"}, self: "raphael", expected: ""},
+		{name: "OnlyOther", peers: []string{"michelangelo"}, self: "raphael", expected: "michelangelo"},
+		{name: "SelfAndOther", peers: []string{"raphael", "michelangelo"}, self: "raphael", expected: "michelangelo"},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			dbPeers := make([]*peers.Peer, 0, len(tc.peers))
+			for _, p := range tc.peers {
+				dbPeers = append(dbPeers, fixtures[p])
+			}
+			db := createDB(t, dbPeers)
+			replica := initReplica(t, db, fixtures[tc.self])
+			selected := replica.SelectPeer()
+			if tc.expected == "" {
+				require.Nil(t, selected, "should not have selected a peer")
+			} else {
+				require.NotNil(t, selected, "should have selected a peer")
+				require.Equal(t, fixtures[tc.expected].Id, selected.Id, "selected the wrong peer")
+			}
+		})
+	}
+}
+
+// Test that peers are selected uniformly at random.
+func TestSelectPeerAll(t *testing.T) {
+	fixtures := loadFixtures(t)
+	peers := make([]*peers.Peer, 0, len(fixtures))
+	for _, p := range fixtures {
+		peers = append(peers, p)
+	}
+	db := createDB(t, peers)
+	replica := initReplica(t, db, fixtures["raphael"])
+
+	// Run enough times so that all peers have a chance to be selected.
+	counter := make(map[uint64]uint8)
+	for i := 0; i < 100; i++ {
+		peer := replica.SelectPeer()
+		require.NotNil(t, peer, "should have selected a peer")
+		counter[peer.Id]++
+	}
+
+	// All peers except the local replica should be selected.
+	require.Len(t, counter, len(peers)-1, "did not select all peers")
```

Comment:
Good use of the fixtures struct in testing - I'm glad this test is setup so we can easily change the peers fixtures!